src/evaluator.ts

```diff
@@ -125,6 +130,19 @@ class Evaluator {
     };
     this.conversations = {};
     this.registers = {};
+    this.gateway = new Gateway({
```

Comment:
If possible, I would like to avoid instantiating a Gateway in evaluator.  This is a hot path that every eval runs through, so provider-specific logic is not ideal.  Could we instantiate in the Gateway provider?